src/Tribe/Views/V2/View.php

```diff
@@ -2194,6 +2194,12 @@ protected function should_show_recent_past_events_view() {
 	 */
 	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
 
+		// Only return if on the current or upcoming view and not in the past.
+		// todo this still displays on day view in the past if it is
+		if ( $template_vars[ 'is_past' ] ) {
```

Comment:
```suggestion
		if ( 'past' === $this->context->get( 'event_display_mode' ) ) {
```

I would use the context as this makes injecting the values much easier.
Relying on a derived value, `$template_vars['is_past']`, means that manipulating this condition will require not only mocking the Context, but also knowing, and hijacking, the logic that uses it to set `$template_vars['is_past']`.

Where possible, as it is here, we should avoid feeding the View its own calculated vars.

Reply:
👍